### PR TITLE
SWC-6644

### PR DESCRIPTION
**Pull request by jay-hodgson**

### CODE REVIEW

**Comment by jay-hodgson** [comment]
In the user react-select component, make the following changes:
1.  Do not pop up the options when the textbox is selected (openMenuOnClick={false})
2.  Instead of the default placeholder text ("Select..."), always set the placeholder text to something meaningful based on the TYPE_FILTER.
3. Only run the useSearchUserGroupHeaders if user input has been defined.

**Comment by jay-hodgson** [comment]
Before:
https://github.com/Sage-Bionetworks/synapse-web-monorepo/assets/1864447/29de2b40-0107-462c-a8a4-34a8ac06923f

After:
https://github.com/Sage-Bionetworks/synapse-web-monorepo/assets/1864447/39807097-1a4c-459a-aeab-0cad7da34924

File: packages/synapse-react-client/src/components/UserSearchBox/UserSearchBoxV2.tsx [hunk 1]
```diff
@@ -135,6 +139,18 @@ const UserSearchBoxV2: React.FC<UserSearchBoxProps> = props => {
...
+    if (placeholder !== undefined) {
+      return placeholder
+    } else if (typeFilter == TYPE_FILTER.USERS_ONLY) {
+      return 'Username or name (first and last)'
+    } else if (typeFilter == TYPE_FILTER.TEAMS_ONLY) {
+      return 'Team name'
+    } else {
+      return 'Username, name (first and last), or team name'
+    }
+  }, [placeholder, typeFilter])
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/UserSearchBox/UserSearchBoxV2.tsx:152]: Nice improvement

File: packages/synapse-react-client/src/components/UserSearchBox/UserSearchBoxV2.tsx [hunk 2]
```diff
@@ -107,9 +107,13 @@ const UserSearchBoxV2: React.FC<UserSearchBoxProps> = props => {
       type: 'USER_PROFILE',
     })
 
+  const isSearchEnabled = !!debouncedInput
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/UserSearchBox/UserSearchBoxV2.tsx:110]: Pretty elegant/simple fix, I like it!

**Review by nickgros — APPROVED** [review]

**Pull request merged**